components/brave_wallet_ui/utils/api-utils.ts

```diff
@@ -29,7 +31,7 @@ export const GetFlattenedAccountBalances = (accounts: WalletAccountType[], userV
         .map(account => {
           const balance = token.contractAddress
             ? account.tokenBalanceRegistry[token.contractAddress.toLowerCase()]
```

Comment:
Are we assuming here that `token.contractAddress` would be different on different chain ids? This would not work for several ERC-20 tokens that are available on multiple networks with the same contract address.

If you create an ERC20 token contract using the `CREATE` opcode:

```
contractAddress = func(deployerAddress, deployerNonce)
// increment deployerNonce until the desired contract address is found
```

If created using `CREATE2`  (newer) opcode:

```
contractAddress = func(deployerAddress, salt, bytecode)
// keep changing salt until the desired contract address is found
```
I think we should namespace the `tokenBalanceRegistry` with `chainId` just like you did for `nativeBalanceRegistry`.

Reply:
Created an issue here https://github.com/brave/brave-browser/issues/21770